### DISCUSSION TITLE
Prolonging session

### DISCUSSION THREAD

**Discussion by RobinBaeckman** [ory/hydra#2684] [Q&A] [answered]
I'm having an usecase where I need to have only one session per user. So user signs in on device A and if they signs in on device B before device As session has expired/terminated they should be blocked. 

Also the session needs to be prolonged every time the refresh token is used (which is done on each request) and also delete the session if the user closes the browser (this is so device B can login right after device A closes their browser)

Are these things possible? How can I prolong the session in hydra on user activity and how can the session be deleted on user browser close?

I'm using the List all consent sessions to check for existing session

**Comment by vinckr** [comment, marked as answer]
Hey @RobinBaeckman ,

OAuth2.0 sessions can get confusing pretty quickly! Check out this document for some info on the various session layers, refresh tokens, and so on: 
https://www.ory.sh/hydra/docs/concepts/before-oauth2#access-and-refresh-tokens-are-not-sessions

What you are looking for is the session that is on your application layer so to speak, so your application needs to store and track and invalidate the session.

Don't use the Ory Hydra Session for this! 

If you want to invalidate the (Ory Hydra) session (i.e. log out the user) just use the [normal flow](https://www.ory.sh/hydra/docs/concepts/logout/).

If you want to invalidate the session in your app, you need to do that there.

**Reply by RobinBaeckman** [reply]
Hi @vinckr, confusing as hell :), it would be so easy to use hydra session instead of creating a whole new session mechanism duplicating what already exists in hydra. 

I'm curious if hydra session even can be prolonged without the user having to re-authenticate? I've been reading hydra docs but can't really find any info that go into depth on what the features of a session are and its lifespan, I found nothing on prolonging it so I assume you can't?

**Reply by vinckr** [reply]
Hella late reply sorry 😓 
I think there is not really documentation on extending the session since there is not much reason to extend it, you just get a new access token anyway and the refresh token is very long lived. 
Also check out this blogpost: https://www.ory.sh/oauth2-openid-connect-do-you-need-use-cases-examples/
